connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/InboundIntermediateConnectorContextImpl.java

```diff
@@ -120,6 +121,16 @@ public void reportHealth(final Health health) {
 
   @Override
   public Health getHealth() {
-    return inboundContext.getHealth();
+    return ((InboundConnectorReportingContext) inboundContext).getHealth();
```

Comment:
Could we get rid of the casting if we apply https://github.com/camunda/connectors/pull/1420/files#r1411746859 ?